### ISSUE TITLE
AriaDescriber should let us describe elements via other DOM Elements (as well as just string)

### ISSUE THREAD

**Issue by mertdeg2** [angular/components#16099]
#### Feature Description

First, i believe this API should be public: https://github.com/angular/components/blob/master/src/cdk/a11y/aria-describer/aria-describer.ts#L70

Second, it can be improved to support describing hostElement (first parameter) via already existing DOM node (instead of just message).

So maybe the new API could look like:

`describe(hostElement: Element, message: string|Element)`

instead of

`describe(hostElement: Element, message: string)`

#### Use Case

In GCP (Google Cloud Platform), we have tried to use this API per @jelbourn recommendation. But we were unable to use it easily because we wanted to provide HTMLElement to the describe function (not a string)
